jina/drivers/encode.py

```diff
@@ -1,25 +1,84 @@
 __copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from . import BaseExecutableDriver
+from typing import Optional
 
-if False:
-    from ..types.sets import DocumentSet
+from . import BaseExecutableDriver
+from ..types.sets import DocumentSet
 
 
 class BaseEncodeDriver(BaseExecutableDriver):
-    """Drivers inherited from this Driver will bind :meth:`craft` by default """
+    """Drivers inherited from this Driver will bind :meth:`encode` by default """
 
-    def __init__(self, executor: str = None, method: str = 'encode', *args, **kwargs):
+    def __init__(self,
+                 executor: str = None,
+                 method: str = 'encode',
+                 *args, **kwargs):
         super().__init__(executor, method, *args, **kwargs)
 
 
 class EncodeDriver(BaseEncodeDriver):
-    """Extract the chunk-level content from documents and call executor and do encoding
+    """Extract the content from documents and call executor and do encoding
     """
 
-    def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        contents, docs_pts, bad_docs = docs.all_contents
+    class CacheDocumentSet:
+        """Helper class to accumulate documents from differents document Set in a single DocumentSet
+         to help guarantee that the encoder driver can consume documents in fixed batch sizes to allow
+         the EncoderExecutors to leverage its batching abilities.
+
+         It is useful to have batching even when chunks are involved"""
+
+        def __init__(self,
+                     capacity: Optional[int] = None,
+                     *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.capacity = capacity
+            self._doc_set = DocumentSet(docs_proto=[])
+
+        @property
+        def available_capacity(self):
+            return self.capacity - len(self._doc_set)
+
+        def cache(self, docs: DocumentSet):
+            docs_to_append = min(len(docs), self.available_capacity)
+            for doc in docs[: docs_to_append]:
+                self._doc_set.append(doc)
```

Comment:
```suggestion
            self._doc_set.extend(docs[: docs_to_append])
```
DocSet has `extend` method